lib/script/taproot.js

```diff
@@ -0,0 +1,92 @@
+/*!
+* taproot.js - taproot interpreter for bcoin
+* https://github.com/bcoin-org/bcoin
+*/
+
+'use strict';
+
+const bio = require('bufio');
+const common = require('./common');
+const { strcmp } = require('../utils/util');
+
+const {
+  TapLeafHash,
+  TapBranchHash
+} = require('../utils/taggedhash');
+
+/**
+ * TapLeaf
+ */
+
+class TapLeaf {
+  /**
+   * Create a leaf in taproot tree.
+   * @constructor
+   * @param {Script} script
+   * @param {Number} leafVersion
+   * @param {String} name
+   */
+
+  constructor(script, leafVersion=common.LEAF_VERSION_TAPSCRIPT, name='') {
+    this.script = script;
+    this.leafVersion = leafVersion;
+    this.name = name;
+  }
```

Comment:
this feels more on-style for us...

We could also assert `instanceOf Script` but that might be overkill

```suggestion
  constructor(script, leafVersion, name) {
    assert(script);
    assert(leafVersion);
    this.script = script;
    this.leafVersion = leafVersion;
    this.name = name || '';
  }
```